examples/wasi_test.go

```diff
@@ -30,22 +30,14 @@ func Test_WASI(t *testing.T) {
 		_, _ = fmt.Fprintf(stdout, "random: %x\n", random)
 	}
 
-	mod, err := wazero.DecodeModuleText([]byte(`(module $wasi
-	(import "wasi_snapshot_preview1" "random_get"
-		(func $wasi.random_get (param $buf i32) (param $buf_len i32) (result (;errno;) i32)))
-	(import "" "random" (func $random))
-	(memory 1)
-	(start $random)
-)`))
-	require.NoError(t, err)
 	store := wazero.NewStore()
 
 	// Host functions can be exported as any module name, including the empty string.
-	_, err = wazero.ExportHostFunctions(store, "", map[string]interface{}{"random": goFunc})
-	require.NoError(t, err)
+	env := &wazero.HostModuleConfig{Name: "", Functions: map[string]interface{}{"random": goFunc}}
+	_, err := wazero.InstantiateHostModule(store, env)
 
 	// Configure WASI and implement the function to use it
-	we, err := wazero.ExportHostFunctions(store, wasi.ModuleSnapshotPreview1, wazero.WASISnapshotPreview1())
```

Comment:
this is one of the more obvious simplifications. At the moment, WASI only exports functions, so we are allowed a simpler config for it!